frontend/src/crosswordView/crosswordView.js

```diff
@@ -0,0 +1,343 @@
+
+        return (
+            <>
+                <Navbar/>
+                <div
+                    className='page'
+                    style={{ 'paddingBottom': '50px' }}
+                >
+                    { showDifficulty
+                        ? <div>
+                            <h1 className='crossword-title'>Select a Crossword Difficulty</h1>
+                            <div className='d-flex justify-content-between mt-5'>
+                                {[EASY, MEDIUM, HARD].map((difficulty) => (
+                                    <button
+                                        className='btn btn-light difficulty-button'
+                                        onClick={() => this.setDifficulty(difficulty)}
+                                        key={difficulty}
+                                    >
+                                        { DIFFICULTY_NAMES[difficulty] }
+                                    </button>
+                                ))}
+                            </div>
+                        </div>
+                        : <>
+                            {instructionModal}
+                            <h1 className='crossword-title'>
+                                Crossword
+                                <button className="btn btn-outline-light btn-circle mx-3"
+                                    style= {{ 'border': '3px solid', 'fontSize': '20px' }}
+                                    onClick={this.modalHandler}>
+                                    <b>?</b>
+                                </button>
+                            </h1>
+                            <button className='btn btn-danger give-up' onClick={this.giveUp}>
+                                Give Up
+                            </button>
+                            <div className='row'>
+                                <div className='col-xl-5 col-12'>
+                                    { clueBox }
+                                    { definitionBox }
+                                </div>
+                                <div
+                                    className='col-xl-7 col-12 d-flex'
+                                    style={{ overflowX: 'scroll', alignContent: 'center' }}
+                                >
+                                    <div className='crossword-grid'>
+                                        <Crossword
+                                            grid={grid}
+                                            isCorrect={this.isCorrect}
+                                            getClueNumber={this.getClueNumber}
+                                            updateCell={this.updateCell}
+                                        />
+                                    </div>
+                                </div>
+                            </div>
+                        </>
+                    }
+
+                </div>
+                <Footer/>
+            </>
+        );
```

Comment:
There's a lot going on in this `render` method, which is making it a bit challenging to grok fully and to skim for things like semantics and accessibility concerns. [This article](https://medium.com/front-end-weekly/react-quick-tip-split-up-your-render-method-5eb30e432e2) outlines a few techniques for keeping your render functions slim, with the general idea being to follow the basic "single-responsibility principle" of programming.